Craigslist

rajatkapoor opened the pull request:
resolves #42 
You will need to add the email credentials in the settings file/ local settings file to test this out.
Please let me know if any further changes are required

owocki commented:
mind posting an example craigslist post?

owocki commented on app/dashboard/notifications.py:
```diff
@@ -223,3 +224,97 @@ def maybe_market_to_email(b, event_name, txid):
+    # submitting final form
+    form = page.soup.form
+    #getting last email id
+    last_email_id = fetch_last_email_id(settings.IMAP_EMAIl, settings.IMAP_PASSWORD)
```
`IMAP_EMAIl`?  shouldnt this be `IMAP_EMAIL`?

rajatkapoor replied:
Typo on my end. Will fix this

owocki commented on app/dashboard/notifications.py:
```diff
@@ -223,3 +224,97 @@ def maybe_market_to_email(b, event_name, txid):
+                page = browser.open(link)
+                form = page.soup.form
+                page = browser.submit(form, form['action'])
+
```
could you have this method return the URL of the post upon success, and `False` upon failure?

rajatkapoor replied:
Sure. Good idea

owocki commented on app/dashboard/notifications.py:
```diff
@@ -223,3 +224,97 @@ def maybe_market_to_email(b, event_name, txid):
 
     return len(to_emails)
 
+def maybe_post_on_craigslist(bounty):
```
`maybe_post_on_craigslist` looks good, but isn't called anywhere.  could you add a management command that posts any bounties with `web3_created` in the last hour to craigslist?

rajatkapoor replied:
Sure. I will also add that

owocki commented on app/dashboard/notifications.py:
```diff
@@ -223,3 +224,97 @@ def maybe_market_to_email(b, event_name, txid):
+    from app.utils import fetch_last_email_id,fetch_mails_since_id
+
+    browser = mechanicalsoup.StatefulBrowser()
+    browser.open(CRAIGSLIST_URL) # open craigslist
```
what locale does `CRAIGSLIST_URL` point to?  i see in the email inbox that these are being posted to bangalore?

owocki replied:
is it possible to enumerate several different locales?

rajatkapoor replied:
We are hitting 'www.craigslist.com' which gets redirects by craigslist. It will get redirected to the locale where our server is hosted. We can enumerate multiple locales as well.

owocki replied:
yes please!

owocki commented:
fwiw, i can't get the method to actually produce a craigslist post..

owocki commented:
https://post.craigslist.org/manage/6379382586/imb8h?lang=en&cc=us

this post seems to have an invalid title / description
<img width="1090" alt="image 2017-11-08 at 5 12 44 pm" src="https://user-images.githubusercontent.com/513929/32581663-0d8dfbea-c4a8-11e7-91ae-606469015213.png">

owocki commented:
just left some requested changes.  this seems to really be coming along and i'm excited.  thank you!

rajatkapoor commented:
I will resolve all these issues and push a fix soon.
The screen shot that you posted is probably for a test post that I did.

owocki commented:
great, ill keep an eye out for your comments.  thanks so much for the progress so far

rajatkapoor commented:
I have fixed the issues that persisted. Also added management command to post to craigslist the bounties created in the last `n` hours:
```
manage.py post_to_craigslist <n>
```

rajatkapoor commented on app/dashboard/notifications.py:
```diff
@@ -277,37 +282,61 @@ def maybe_post_on_craigslist(bounty):
+                    return link
+                except:
+                    # in case of inavalid links
+                    pass
```
Absolutely correct. Will fix this

rajatkapoor replied:
Actually, this is fine.
The regex here is a slightly 'loose' regex, so it can match some links which might be invalid. In case of an invalid link we just move ahead and look for further links. We return the link whenever we come across a valid one. At the end, when the loop is completed without us breaking it or returning it again, that means we did not find any valid link. Hence we return false in the 'else; clause

rajatkapoor commented on app/dashboard/notifications.py:
```diff
@@ -223,3 +224,126 @@ def maybe_market_to_email(b, event_name, txid):
+    # Final form filling
+    form.find('input', {'id': "PostingTitle"})['value'] = posting_title
+    form.find('textarea', {'id': "PostingBody"}).insert(0, posting_body)
+    form.find('input', {'id': "FromEMail"})['value'] = settings.CONTACT_EMAIL
```
It has to be the IMAP_EMAIL only. Will fix this as well

rajatkapoor commented on app/dashboard/notifications.py:
```diff
@@ -277,37 +282,61 @@ def maybe_post_on_craigslist(bounty):
     form.find('input', {'value': 'pay', 'name': 'remuneration_type'})['checked'] = ''
-    form.find('input', {'id': "remuneration"})['value'] = "{} ETH or {} USD".format(bounty.value_in_eth, bounty.value_in_usdt)
-
+    form.find('input', {'id': "remuneration"})['value'] = "{} ETH".format(bounty.value_in_eth/10**18)
```
Thanks for the input. Such bugs will be taken care of once I gain more familiarity with the code base.  Will fix this for now

owocki commented:
i keep getting `None` back for the link URLs

```
kevinowocki@local /Users/kevinowocki/Sites/gitcoin/web/app~ % ./manage.py post_to_craigslist 100                  (git)-[craigslist] :
/Library/Python/2.7/site-packages/rest_framework/routers.py:23: RemovedInDjango20Warning: Importing from django.core.urlresolvers is deprecated in favor of django.urls.
  from django.core.urlresolvers import NoReverseMatch
Posted None
Posted None
Posted None
```

owocki commented:
what kind out output do you see when you run the script?  can you send me a link to any of the craigslist posts your script generated?

rajatkapoor commented:
<img width="1437" alt="screen shot 2017-11-13 at 11 34 30 pm" src="https://user-images.githubusercontent.com/1975189/32741207-6b112e14-c8cb-11e7-81b3-4408919a4440.png">
Here is a screenshot of the output that I get

This is the url in the image: `https://post.craigslist.org/u/8DDe7ZzI5xGg6x-pKlIXIQ/492br`

owocki commented:
can you try running against another location (boulder colorado USA instead of bangalore)?   for some reason it's not working for me, and i wonder if its the geo-routing of craigslist

rajatkapoor commented:
Made all the requested changes. Please have a look
